### ISSUE TITLE
Armor Math bug

### ISSUE THREAD

**Issue by james-rae** [grousewood-games/solforge#97]
The following wrong scenario can occur.

- Creature has Armor 2 due to Nexus Tech being on board
- Creature receives 2 damage. Armor absorbs it.
- Nexus Tech dies. Creature loses the (spent) 2 Armor
- Tech Upgrade is cast on the creature, giving it 2 Armor
- Creature receives 2 damage.
- Would expect the "new" Armor to absorb this as well. It will not. The creature still considers itself to have used up 2 of its Armor capacity.

Basically any Armor that can disappear mid-turn (and is not Negated) will be effected by this scenario.

Fix is to unwind any Armor damage when removing Armor (without going negative).
